projects/common/src/lib/components/iot-gateway-endpoint/ige-resources/ige-resources.component.ts

```diff
@@ -129,21 +129,21 @@ export class IgeResourcesComponent implements OnInit {
   onFileUpload(input: Event) {
     const inputElement = input.target as HTMLInputElement;
     const file = inputElement.files.item(0);
-    // if (file.type && file.type == 'text/x-python-script') {
-    //   this.resourceForm.patchValue({ type: 'processor' });
-    //   this.readFile(inputElement);
-    // } else if (file.name.match('(?:^|\W)dockerfile(?:$|\W)')) {
-    //   this.resourceForm.patchValue({ type: 'dockerfile' });
-    //   this.readFile(inputElement);
-    // } else if (file.name == "requirement.txt") {
-    //   this.resourceForm.patchValue({ type: 'requirement'});
-    //   this.readFile(inputElement);
-    // } else {
-    //   this._snackBar.open('Resource file is not supported.', 'Cancel', {
-    //     duration: 3000,
-    // });
-    // }
-    this.readFile(inputElement);
+    if (file.type && file.type == ('text/x-python-script') || ('text/x-python')) {
+      this.resourceForm.patchValue({ type: 'processor' });
+      this.readFile(inputElement);
+    } else if (file.name.match('(?:^|\W)dockerfile(?:$|\W)')) {
+      this.resourceForm.patchValue({ type: 'dockerfile' });
+      this.readFile(inputElement);
+    } else if (file.name == "requirement.txt") {
+      this.resourceForm.patchValue({ type: 'requirement'});
+      this.readFile(inputElement);
+    } else {
+      this.readFile(inputElement);
+      this._snackBar.open('Resource file may not be supported.', 'Cancel', {
+        duration: 3000,
+    });
+    }
```

Comment:
Can you put a unit test in place for these conditions? It might require a small refactor to pass in the file object to a method if mocking the `inputElement.files.item(0)` proves to be difficult. See the link below for a hint.

I think the test would be good because I am not sure that the regular expressions are right. I know this is an uncommenting out of what might be legacy code, but is `'(?:^|\W)dockerfile(?:$|\W)'` correct for matching `Dockerfile`? I normally use a capital 'D' and give optional '-example' on the end.

If you can add this test in a few minutes, I think it might prove to be useful.

https://stackoverflow.com/questions/51970681/angular-jasmine-unit-test-change-event-for-inputtype-file

Reply:
Yes, absolutely.